Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.cpp

```diff
@@ -552,13 +588,29 @@ void HybridPICModel::FieldPush (
 {
     auto& warpx = WarpX::GetInstance();
 
+    amrex::Real const t_old = warpx.gett_old(0);
+
     // Calculate J = curl x B / mu0 - J_ext
     CalculatePlasmaCurrent(Bfield, eb_update_E);
     // Calculate the E-field from Ohm's law
     HybridPICSolveE(Efield, Jfield, Bfield, rhofield, eb_update_E, true);
     warpx.FillBoundaryE(ng, nodal_sync);
+    warpx.ApplyEfieldBoundary(0, PatchType::fine, t_old);
+
     // Push forward the B-field using Faraday's law
-    amrex::Real const t_old = warpx.gett_old(0);
     warpx.EvolveB(dt, dt_type, t_old);
     warpx.FillBoundaryB(ng, nodal_sync);
+    warpx.ApplyBfieldBoundary(0, PatchType::fine, dt_type, t_old);
+}
+
+void
+WarpX::CalculateExternalCurlA() {
```

Comment:
I still need to review the rest to understand why this has to be a member function of `WarpX`, but in any case it would probably be good to move it to a different file, perhaps `WarpXPushFieldsHybridPIC.cpp`.

Reply:
This is a hook that I added to expose CalculateExternalCurlA as a static function to be called in a python callback. It is used to load fields dynamically in the MFs then this is called at the end of the callback to populate the external E and B arrays from A. It is more of an advanced use case and I used it for initial debugging, but could probably be removed if you think it is too messy to keep it around. I will likely use it again when integrating lattice elements into this framework.

Reply:
I think it is fine to have, I just think it should live in a file that generally defines `WarpX` member functions, so I think moving it to `WarpXPushFieldsHybridPIC.cpp` would be a fine change.